src/main/ruby/truffleruby/core/dir_glob.rb

```diff
@@ -245,13 +245,15 @@ def process_directory(matches, parent, entry, glob_base_dir)
             type = entry.type
 
             full = path_join(path, ent)
-            if (type == Truffle::DirOperations::DT_DIR) and (allow_dots or ent.getbyte(0) != 46) # ?.
-              @next.process_entry ent, type, matches, path, glob_base_dir
-
-              stack << [path, dir_entries]
-              path = full
-              dir_entries = subdir_entries(glob_base_dir, full)
-            elsif (allow_dots or ent.getbyte(0) != 46) # ?.
+            if (type == Truffle::DirOperations::DT_DIR)
+              if (allow_dots or ent.getbyte(0) != 46) # ?.
```

Comment:
Minor, but we should remove the `()` since they group the entire condition here and so are redundant

Reply:
Let's fix that later to not delay this PR

Reply:
You're right, I split that line without updating the style.
i will push a follow up PR.
There is another place we could make a similar change (for a different subset of glob patterns) that I already have some specs ready for.